packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.derived.js

```diff
@@ -1,340 +1,392 @@
+    //     <SearchBox defaultRefinement="second query 1" />
+    //   </Index>
+    // </InstantSearch>
+
+    const ism = createInstantSearchManager({
+      indexName: 'first',
+      initialState: {},
+      searchParameters: {},
+      searchClient: createSearchClient(),
+    });
 
-describe('createInstantSearchManager', () => {
-  describe('with correct result from algolia', () => {
-    describe('on widget lifecycle', () => {
-      it('updates the store and searches', () => {
-        expect.assertions(7);
+    // <SearchBox defaultRefinement="first query 1" />
+    ism.widgetsManager.registerWidget({
+      getSearchParameters: params => params.setQuery('first query 1'),
+      context: {},
+      props: {},
+    });
 
-        const ism = createInstantSearchManager({
-          indexName: 'first',
-          initialState: {},
-          searchParameters: {},
-          searchClient: client,
-        });
+    // <Index indexName="first" />
+    ism.widgetsManager.registerWidget({
+      getSearchParameters: params => params.setIndex('first'),
+      context: {},
+      props: {
+        indexName: 'first',
+      },
+    });
 
-        ism.widgetsManager.registerWidget({
-          getSearchParameters: params => params.setQuery('first query 1'),
-          context: {},
-          props: {},
-        });
+    // <Index indexName="first">
+    //   <Pagination defaultRefinement={3} />
+    // </Index>
+    ism.widgetsManager.registerWidget({
+      getSearchParameters: params => params.setPage(3),
+      context: { multiIndexContext: { targetedIndex: 'first' } },
+      props: {},
+    });
 
-        ism.widgetsManager.registerWidget({
-          getSearchParameters: params => params.setQuery('second query 1'),
-          context: { multiIndexContext: { targetedIndex: 'second' } },
-          props: {},
-        });
+    // <Index indexName="second" />
+    ism.widgetsManager.registerWidget({
+      getSearchParameters: params => params.setIndex('second'),
+      context: {},
+      props: {
+        indexName: 'second',
+      },
+    });
 
-        ism.widgetsManager.registerWidget({
-          getSearchParameters: params => params.setPage(3),
-          context: { multiIndexContext: { targetedIndex: 'first' } },
-          props: {},
-        });
+    // <Index indexName="second">
+    //   <SearchBox defaultRefinement="second query 1" />
+    // </Index>
+    ism.widgetsManager.registerWidget({
+      getSearchParameters: params => params.setQuery('second query 1'),
+      context: { multiIndexContext: { targetedIndex: 'second' } },
+      props: {},
+    });
 
-        ism.widgetsManager.registerWidget({
-          getSearchParameters: params => params.setIndex('second'),
-          context: { multiIndexContext: { targetedIndex: 'second' } },
-          props: { indexName: 'second' },
-        });
+    expect(ism.store.getState().results).toBe(null);
+
+    return Promise.resolve()
+      .then(() => {})
```

Comment:
Why do you need this?

Reply:
Basically every time a `registerWidget` is called [we scheduled an update](url) on the next tick. The manager does not return the promise so we have no ways to execute an instruction once the delay is elapsed. That way we "wait" the next tick. The idea is similar to what Jest does for fake timers. There is [some discussions](https://github.com/facebook/jest/issues/2157) to add this feature to Jest.